electrumsv/transaction_store.py

```diff
@@ -0,0 +1,78 @@
+import os
+from typing import Any
+
+from .bitcoin import sha256d
+from .app_state import app_state
+from .transaction import Transaction
+
+class TransactionStore:
+    def __init__(self, keys) -> None:
+        self._keys = frozenset(keys)
+        self._cache = {}
+        self._cache_path = os.path.join(app_state.config.electrum_path(), "txcache")
+        if not os.path.exists(self._cache_path):
+            os.makedirs(self._cache_path)
+
+    def _is_persisted(self, tx_id: str) -> bool:
+        txcache_path = os.path.join(self._cache_path, tx_id)
+        return os.path.exists(txcache_path)
+
+    def keys(self):
+        return self._keys
+
+    def items(self):
+        raise NotImplementedError
+
+    def get(self, tx_id: str, default: Any=None) -> Transaction:
+        if tx_id in self._keys:
+            return self[tx_id]
+        return default
+
+    def __iter__(self):
+        return iter(self._keys)
+
+    def __contains__(self, tx_id: str) -> bool:
+        assert len(tx_id) == 32*2
+        if tx_id in self._cache:
+            return True
+        return tx_id in self._keys
```

Comment:
I would expect the conditions for get() to return default and __contains__ to return False would be identical.